src/common/crypto/mbedtls/driverwrapper.cpp

```diff
@@ -421,45 +448,50 @@ psa_status_t aos_get_builtin_key(psa_drv_slot_number_t slotNumber, psa_key_attri
 
-                break;
+        return PSA_ERROR_INVALID_ARGUMENT;
+    }
 
-            case aos::crypto::KeyTypeEnum::eECDSA: {
-                psa_set_key_algorithm(attributes, PSA_ALG_ECDSA(sPSAAlgs[static_cast<int>(key.mHashAlg)]));
-                auto oid = static_cast<const aos::crypto::ECDSAPublicKey&>(key.mPrivKey->GetPublic()).GetECParamsOID();
-                auto curveParameters = FindPsaECGroupByOID(oid);
-                if (curveParameters.mSecond == 0) {
-                    LOG_ERR() << "EC group not found: slotNumber = " << slotNumber;
+    switch (sBuiltinKeys[slotNumber].mPrivKey->GetPublic().GetKeyType().GetValue()) {
+    case aos::crypto::KeyTypeEnum::eRSA:
+        psa_set_key_type(attributes, PSA_KEY_TYPE_RSA_KEY_PAIR);
+        psa_set_key_algorithm(
+            attributes, PSA_ALG_RSA_PKCS1V15_SIGN(sPSAAlgs[static_cast<int>(sBuiltinKeys[slotNumber].mHashAlg)]));
 
-                    return PSA_ERROR_NOT_SUPPORTED;
-                }
+        break;
 
-                psa_set_key_type(attributes, PSA_KEY_TYPE_ECC_KEY_PAIR(curveParameters.mFirst));
-                psa_set_key_bits(attributes, curveParameters.mSecond);
+    case aos::crypto::KeyTypeEnum::eECDSA: {
+        psa_set_key_algorithm(attributes, PSA_ALG_ECDSA(sPSAAlgs[static_cast<int>(sBuiltinKeys[slotNumber].mHashAlg)]));
+        auto oid = static_cast<const aos::crypto::ECDSAPublicKey&>(sBuiltinKeys[slotNumber].mPrivKey->GetPublic())
+                       .GetECParamsOID();
+        auto curveParameters = FindPsaECGroupByOID(oid);
+        if (curveParameters.mSecond == 0) {
+            LOG_ERR() << "EC group not found: slotNumber = " << slotNumber;
 
-                break;
-            }
+            return PSA_ERROR_NOT_SUPPORTED;
+        }
 
-            default:
-                LOG_ERR() << "Not supported key type: keyType = " << key.mPrivKey->GetPublic().GetKeyType();
+        psa_set_key_type(attributes, PSA_KEY_TYPE_ECC_KEY_PAIR(curveParameters.mFirst));
+        psa_set_key_bits(attributes, curveParameters.mSecond);
 
-                return PSA_ERROR_NOT_SUPPORTED;
-            }
+        break;
+    }
 
-            psa_set_key_id(attributes, key.mKeyID);
-            psa_set_key_lifetime(attributes, key.mLifetime);
-            psa_set_key_usage_flags(attributes, PSA_KEY_USAGE_SIGN_HASH | PSA_KEY_USAGE_VERIFY_HASH);
+    default:
+        LOG_ERR() << "Not supported key type: keyType = "
+                  << sBuiltinKeys[slotNumber].mPrivKey->GetPublic().GetKeyType();
 
-            return keyBufferSize != 0 ? PSA_SUCCESS : PSA_ERROR_BUFFER_TOO_SMALL;
-        }
+        return PSA_ERROR_NOT_SUPPORTED;
     }
 
+    psa_set_key_id(attributes, sBuiltinKeys[slotNumber].mKeyID);
+    psa_set_key_lifetime(attributes, sBuiltinKeys[slotNumber].mLifetime);
+    psa_set_key_usage_flags(attributes, PSA_KEY_USAGE_SIGN_HASH | PSA_KEY_USAGE_VERIFY_HASH);
+
+    return keyBufferSize != 0 ? PSA_SUCCESS : PSA_ERROR_BUFFER_TOO_SMALL;
```

Comment:
looks like method's code below return can be removed

Reply:
Done